widgets/src/main/java/dev/chrisbanes/insetter/widgets/constraintlayout/InsetterConstraintHelper.java

```diff
@@ -169,128 +163,115 @@ public WindowInsets onApplyWindowInsets(WindowInsets insets) {
       View view = container.getViewById(mIds[i]);
       final ViewState state = (ViewState) view.getTag(R.id.insetter_initial_state);
       if (state != null) {
-        Insetter.applyInsetsToView(
-            view,
-            insetsCompat,
-            state,
-            systemWindowInsetsPaddingDimensions,
-            systemWindowInsetsMarginDimensions,
-            systemGestureInsetsPaddingDimensions,
-            systemGestureInsetsMarginDimensions);
+        Insetter.builder()
+            .applySystemWindowInsetsToPadding(systemWindowInsetsPaddingSides)
+            .applySystemWindowInsetsToMargin(systemWindowInsetsMarginSides)
+            .applySystemGestureInsetsToPadding(systemGestureInsetsPaddingSides)
+            .applySystemGestureInsetsToMargin(systemGestureInsetsMarginSides)
+            .build()
+            .applyInsetsToView(view, insetsCompat, state);
       }
     }
 
     return insetsCompat.toWindowInsets();
   }
 
   /**
-   * Returns the {@link EnumSet} of {@link InsetDimension}s which define which padding dimensions
-   * should be applied using the system window insets.
+   * Returns the sides on which system window insets should be applied to the padding.
    *
    * <p>This value can be set using the {@code app:paddingSystemWindowInsets} attribute.
    *
    * @see WindowInsetsCompat#getSystemWindowInsets()
    */
-  @Nullable
-  public EnumSet<InsetDimension> getSystemWindowInsetsPaddingDimensions() {
-    return systemWindowInsetsPaddingDimensions;
+  public int getSystemWindowInsetsPaddingSides() {
+    return systemWindowInsetsPaddingSides;
   }
 
   /**
-   * Set the {@link InsetDimension}s which define which padding dimensions should be applied using
-   * the system window insets.
+   * Set the sides on which system window insets should be applied to the padding.
    *
    * <p>This value can be set using the {@code app:paddingSystemWindowInsets} attribute.
    *
    * @see WindowInsetsCompat#getSystemWindowInsets()
    */
-  public void setSystemWindowInsetsPaddingDimensions(@Nullable EnumSet<InsetDimension> values) {
-    if (!ObjectsCompat.equals(systemWindowInsetsPaddingDimensions, values)) {
-      systemWindowInsetsPaddingDimensions = values;
+  public void setSystemWindowInsetsPaddingSides(@Sides int flags) {
+    if (!ObjectsCompat.equals(systemWindowInsetsPaddingSides, flags)) {
```

Comment:
I guess this could just be an `int` comparison now?

Reply:
Yep, perfect!